docs/how-to/download-new-upstream-version.rst

```diff
@@ -1,9 +1,139 @@
 ..  _download-new-upstream-version:
 
-===============================
-Download a new Upstream Version
-===============================
+Download a new :term:`Upstream` version
+=======================================
 
 .. caution::
 
     Work in progress!
+
+Once in a while you need to download a new :term:`Upstream` release or check if a newer
+:term:`Upstream` release exists.
+
+Most of the :term:`Source Packages <Source Package>` contain a ``watch`` file in the ``debian``
+folder. This is a configuration file for the :manpage:`uscan(1)` utility which allows you to
+automatically search :term:`HTTP` or :term:`FTP` sites or :manpage:`git(1)` repositories
+for newly available updates of the :term:`Upstream` project.
+
+.. note::
+    If the :term:`Source Package` does not contain a :file:`debian/watch` file, there may be
+    an explanation and instructions in the :file:`debain/README.source` or
+    :file:`debian/README.debian` file (if available) that tell you how to proceed.
+
+.. caution::
+    You should download :term:`Upstream` file(s) manually only if there is no automatic
+    download mechanism and you can't find any download instructions.
+
+    Remember that a package may get distributed to hundreds of thousands of users.
+    Humans are the weakest link in this distribution chain, because we may
+    accidentally miss or skip a verification step, misspell a :term:`URL`, copy the
+    wrong :term:`URL` or copy a :term:`URL` only partially, etc.
+
+    If you still have to download :term:`Upstream` file(s) manually make sure to
+    verify :term:`Cryptographic Signatures` (if available). The :term:`Signing Key`
+    of the :term:`Upstream` project should be stored in the :term:`Source Package`
+    as :file:`debian/upstream/signing-key.asc` (if the :term:`Upstream` project
+    has a :term:`Signing Key`).
+
+    :manpage:`uscan(1)` verifies downloads against this :term:`Signing Key`
+    automatically (if available).
+
+Download new :term:`Upstream` version (if available)
+----------------------------------------------------
+
+Running :manpage:`uscan(1)` from the :term:`Root` of the :term:`Source Tree` will
+check if a newer :term:`Upstream` version exits and downloads it:
+
+.. code:: bash
+
+    uscan
+
+If :manpage:`uscan(1)` could not find a newer :term:`Upstream` version it will
+return with the exit code `1` and print nothing to the :term:`Standard Output`.
+
+:manpage:`uscan(1)` reads the first entry in :file:`debian/changelog` to determine
+the name and version of the :term:`Source Package`.
+
+You can allways add the ``--verbose`` flag to see more information (e.g., which version
+:manpage:`uscan(1)` found):
+
+.. code:: bash
+
+    uscan --verbose
+
+Check for new :term:`Upstream` version (no download)
```

Comment:
I'd suggest we move this to be the first sub-heading. Any user may want to check their version before downloading, so I think that would be a more logical ordering.

Reply:
You don't have to run `uscan --safe` to run `uscan` the `--safe` flag basically disables downloading anything. Both are for 2 different use cases.

You would run `uscan`, e.g.:
- as the package maintainer to download and package the new release

You would run `uscan --safe`, e.g.:
- during a [Main Inclusion Review](https://github.com/canonical/ubuntu-mir) just to check if the `debian/watch` file works
- [while fixing a bug](https://github.com/canonical/ubuntu-maintainers-handbook/blob/f788e02eccb4a7cf9696585e4e7634d8d2735ed7/PackageFixing.md#was-it-fixed-upstream) just to make sure that there is no newer version that may have fixed the bug already

Reply:
@s-makin With that said; do you still think that the order should be moved?

Reply:
My impression is that checking the debian/watch file works or fixing a bug might be easier or at least more common steps that a packager would need to know before they become a package maintainer (although I could be wrong there!). If that's correct, then I'd move the `uscan --safe` section. Otherwise I'd leave it as-is - I'll bow to your greater knowledge on this one!

Reply:
Actually I don't know which use case is more common. I will ask around in `Foundations`